include/fluent-bit/flb_aws_credentials.h

```diff
@@ -131,6 +131,13 @@ struct flb_aws_provider {
 
     /* Standard credentials chain is a list of providers */
     struct mk_list _head;
+
+    /* Provider managed dependencies; to delete on destroy */
+    struct flb_aws_provider *base_aws_provider;
+    struct flb_tls *cred_tls;   /* tls instances can't be re-used; aws provider requires
+                                   a separate one */
+    struct flb_tls *sts_tls;    /* one for the standard chain provider, one for sts
+                                   assume role */
```

Comment:
Hmmm this is only used by the new `flb_managed_chain_provider_create` function to use? the other providers don't use these new fields I guess?

Reply:
That's right. It's just if we want aws_provider_destroy to handle clean up operations. If null (due to calloc) clean up will not occur